demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1051 @@
+                {
+                    LogError( ( "Error getting the property version" ) );
+                    break;
+                }
+
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+            }
+            else
+            {
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                xResult = AzureIoTJSONReader_SkipChildren( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+
+                xResult = AzureIoTJSONReader_NextToken( &xReader );
+                configASSERT( xResult == eAzureIoTSuccess );
+            }
+        }
+
+        if( xResult != eAzureIoTErrorEndOfProperties )
+        {
+            LogError( ( "There was an error parsing the properties: 0x%08x", xResult ) );
+        }
+        else
+        {
+            LogInfo( ( "Successfully parsed properties" ) );
+            xResult = eAzureIoTSuccess;
+        }
+    }
+
+    return xResult;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Update local device temperature values based on new requested temperature.
+ */
+static void prvUpdateLocalProperties( double xNewTemperatureValue,
+                                      uint32_t ulPropertyVersion,
+                                      bool * pxOutMaxTempChanged )
+{
+    *pxOutMaxTempChanged = false;
+    xDeviceCurrentTemperature = xNewTemperatureValue;
+
+    /* Update maximum or minimum temperatures. */
+    if( xDeviceCurrentTemperature > xDeviceMaximumTemperature )
+    {
+        xDeviceMaximumTemperature = xDeviceCurrentTemperature;
+        *pxOutMaxTempChanged = true;
+    }
+    else if( xDeviceCurrentTemperature < xDeviceMinimumTemperature )
+    {
+        xDeviceMinimumTemperature = xDeviceCurrentTemperature;
+    }
+
+    /* Calculate the new average temperature. */
+    ulDeviceTemperatureCount++;
+    xDeviceTemperatureSummation += xDeviceCurrentTemperature;
+    xDeviceAverageTemperature = xDeviceTemperatureSummation / ulDeviceTemperatureCount;
+
+    LogInfo( ( "Client updated desired temperature variables locally." ) );
+    LogInfo( ( "Current Temperature: %2f", xDeviceCurrentTemperature ) );
+    LogInfo( ( "Maximum Temperature: %2f", xDeviceMaximumTemperature ) );
+    LogInfo( ( "Minimum Temperature: %2f", xDeviceMinimumTemperature ) );
+    LogInfo( ( "Average Temperature: %2f", xDeviceAverageTemperature ) );
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Send updated maximum temperature value to IoT Hub.
+ */
+static void prvSendNewMaxTemp( double xUpdatedTemperature )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONWriter_t xWriter;
+    int32_t lBytesWritten;
+
+    xResult = AzureIoTJSONWriter_Init( &xWriter, ucPropertyPayloadBuffer, sizeof( ucPropertyPayloadBuffer ) );
```

Comment:
Again per my comments above on other JSONWriter_Init, recomend some commentary around this.